sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java

```diff
@@ -0,0 +1,216 @@
+        this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+        scheduledExecutor.scheduleWithFixedDelay(() -> {
+            if (this.asyncClient.get().isConnectionClosed()) {
+                restartMessageReceiver();
+            }
+        }, SCHEDULER_INTERVAL_IN_SECONDS, SCHEDULER_INTERVAL_IN_SECONDS, TimeUnit.SECONDS);
+
+    }
+
+    /**
+     * Stops the message processing for this processor. The receiving links and sessions are kept active and this
+     * processor can resume processing messages by calling {@link #start()} again.
+     */
+    public synchronized void stop() {
+        isRunning.set(false);
+    }
+
+    /**
+     * Stops message processing and closes the processor. The receiving links and sessions are closed and calling
+     * {@link #start()} will create a new processing cycle with new links and new sessions.
+     */
+    public synchronized void close() {
+        isRunning.set(false);
+        if (receiverSubscription.get() != null) {
+            receiverSubscription.get().cancel();
+        }
+        asyncClient.get().close();
+        scheduledExecutor.shutdown();
+    }
+
+    /**
+     * Returns {@code true} if the processor is running. If the processor is stopped or closed, this method returns
+     * {@code false}.
+     *
+     * @return {@code true} if the processor is running.
+     */
+    public synchronized boolean isRunning() {
+        return isRunning.get();
+    }
+
+    private synchronized void receiveMessages() {
+        if (receiverSubscription.get() != null) {
+            receiverSubscription.get().request(1);
+            return;
+        }
+        ServiceBusReceiverAsyncClient receiverClient = asyncClient.get();
+        receiverClient.receiveMessages()
+            .parallel(processorOptions.getMaxConcurrentCalls())
+            .runOn(Schedulers.boundedElastic())
+            .subscribe(new Subscriber<ServiceBusReceivedMessageContext>() {
+                @Override
+                public void onSubscribe(Subscription subscription) {
+                    receiverSubscription.set(subscription);
+                    receiverSubscription.get().request(1);
+                }
+
+                @Override
+                public void onNext(ServiceBusReceivedMessageContext serviceBusReceivedMessageContext) {
+                    if (serviceBusReceivedMessageContext.hasError()) {
+                        handleError(serviceBusReceivedMessageContext.getThrowable());
+                    } else {
+                        try {
+                            ServiceBusProcessorMessageContext serviceBusProcessorMessageContext =
+                                new ServiceBusProcessorMessageContext(receiverClient, serviceBusReceivedMessageContext);
+                            processMessage.accept(serviceBusProcessorMessageContext);
+                        } catch (Exception exception) {
```

Comment:
If  user's `processHandler` throw error when they want to abandon the message , are we abandoning the message ?

Reply:
Yes, we should abandon the message if user's message handler throws an exception. Updated to abandon the message.